crates/swc/src/builder.rs

```diff
@@ -180,16 +180,19 @@ impl<'a, 'b, P: swc_ecma_visit::Fold> PassBuilder<'a, 'b, P> {
                 self.top_level_mark,
                 comments,
                 env,
+                self.assumptions,
             ))
         } else {
+            let assumptions = self.assumptions;
             Either::Right(chain!(
                 Optional::new(
                     compat::es2022::es2022(compat::es2022::Config {
                         class_properties: compat::es2022::class_properties::Config {
-                            private_as_properties: self.loose,
-                            constant_super: self.loose,
-                            set_public_fields: self.loose,
-                            no_document_all: self.loose
+                            private_as_properties: self.loose
```

Comment:
wouldn’t it be better to modify `assumptions` according to `loose` setting and pass down only the `assumptions`? (referring also to changes in `swc_ecma_preset_env`; something like this is already done in [config/mod.rs](https://github.com/swc-project/swc/pull/3766/files#diff-05ec577011f6096aac154c266838a96b40e857713f31a3441fc73baf1d90580e)?)

Reply:
Yes, `self.loose` here is redundant